andela/ah-frontend-poseidon
# #162162977: Add Search Functionality and Custom filtering based on parameters.

**What does this PR do?**
- Add search functionality to the application such that users can search articles based on parameters like author, title, tags, keyword.

**Description of tasks to be completed?**
- Add search bar to NavBarComponent
- Add dropdown form for filter options
- Implement functionality through event handling methods.

**How can this be manually tested?**
- Clone the repository
- git checkout ft-search-functionality-162162977
- Install dependencies using yarn install
- Run the app using yarn start
- The application should be running.
- Open the browser and go to http://localhost:3000
- Do several searches using the `search bar` and `search by filters`.

**Screenshots?**
![image](https://user-images.githubusercontent.com/40595048/51993705-69de2e00-24c0-11e9-9fcd-7faccdc49267.png)

![image](https://user-images.githubusercontent.com/40595048/51993730-76628680-24c0-11e9-8f7d-e52cc1a1f88d.png)

![image](https://user-images.githubusercontent.com/40595048/51993756-84180c00-24c0-11e9-9b4d-c9a9f86862a6.png)

![image](https://user-images.githubusercontent.com/40595048/52045418-8b3d2980-2555-11e9-96ae-7b3a86372870.png)

**What are the related user stories?**
[#162162977]
